test/new-e2e/pkg/utils/e2e/client/agent_host.go

```diff
@@ -19,10 +20,24 @@ type agentHostExecutor struct {
 }
 
 func newAgentHostExecutor(host *components.RemoteHost) agentCommandExecutor {
+	var err error
+	var path string
+	switch host.OSFamily {
+	case os.WindowsFamily:
+		// If the agent is installed get the path from the registry, fallback to default path
+		path, err = windowsAgent.GetInstallPathFromRegistry(host)
+		if err != nil {
+			path = windowsAgent.DefaultInstallPath
+		}
+	}
+	return newAgentHostExecutorWithInstallPath(host, path)
+}
+
+func newAgentHostExecutorWithInstallPath(host *components.RemoteHost, installPath string) agentCommandExecutor {
```

Comment:
<!-- Warning highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes and they are blockers. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question. -->
**🔨 warning**
`installPath` seems like it is only allowed on windows, would refactor `newAgentHostExecutor` to create the `baseCommand` and have this function taking the `baseCommand` as input, rather than a param that is unused